src/App.tsx

```diff
@@ -14,11 +15,103 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
+
+enum Operation {
+  Add,
+  Remove,
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: ['Jam'],
+  };
+
+  goodsOperationsHandler = (operation: Operation, good: string) => {
+    switch (operation) {
+      case Operation.Add:
+        this.setState((state) => ({ selectedGoods: [...state.selectedGoods, good] }));
+
+        break;
+      case Operation.Remove:
+        this.setState((state) => (
+          { selectedGoods: state.selectedGoods.filter(goodItem => goodItem !== good) }
+        ));
+
+        break;
+      default:
+    }
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+    const goodsLengthCheck = selectedGoods.length > 1
+      ? `${selectedGoods.slice(0, -1).join(', ')} and ${selectedGoods[selectedGoods.length - 1]} are selected`
+      : `${selectedGoods[0]} is selected`;
+
+    return (
+      <div className="App">
+        <div className="App__top">
+          <h1>
+            {selectedGoods.length ? goodsLengthCheck : 'No goods selected'}
+          </h1>
+          {!!selectedGoods.length && (
+            <button
+              type="button"
+              onClick={() => {
+                this.setState({ selectedGoods: [] });
+              }}
+              onKeyDown={() => {
+                this.setState({ selectedGoods: [] });
+              }}
+            >
+              X
+            </button>
+          )}
+        </div>
+        <ul>
+          {goodsFromServer.map((good) => (
+            <li key={good}>
+              <div className="App__good">
+                <p className={classNames('App__good-text', { 'App__good-text-selected': selectedGoods.includes(good) })}>
```

Comment:
If `selected` is supposed to be a modifier it is advisable that you put double hyphen before stating it, so it is more obvious)
```suggestion
                <p className={classNames('App__good-text', { 'App__good-text--selected': selectedGoods.includes(good) })}>
```

Reply:
Thanks for noticing, you actually got me 2nd time on this, i should really be more patient to my classes)